app/core/exceptions/grading.py

```diff
@@ -3,4 +3,19 @@
 class OtterConfigViolationException(CustomException):
     code = 403
     error_code = "GRADING__OTTER_CONFIG_VIOLATION"
-    message = "otterconfig is invalid or cannot be generated for assignment"
\ No newline at end of file
+    message = "otterconfig is invalid or cannot be generated for the assignment"
+
+class AutogradingDisabledException(CustomException):
+    code = 400
+    error_code = "GRADING__AUTOGRADING_DISABLED"
+    message = "autograding has been disabled for the assignment"
+
+class StudentGradedMultipleTimesException(CustomException):
+    code = 400
+    error_code = "GRADING__STUDENT_GRADED_MULTIPLE_TIMES"
+    message = "only 1 submission per student is permitted for batch grading"
```

Comment:
I would use a term here other than "batch grading" if you expect that this error might ever be shown to users. It's kind of an esoteric term for the lay person, and the message was kind of confusing for me too at first. What about "only one submission per student is permitted during grading"? We don't have to stipulate "batch grading" at all since the scenario where we grade only one submission total precludes the possibility of attempting to grade multiple submissions from one student.